GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs

```diff
@@ -84,29 +86,48 @@ private void UpdateMyRepos()
             _myReposLV.Items.Clear();
             _myReposLV.Items.Add(new ListViewItem { Text = _strLoading.Text });
 
-            AsyncLoader.DoAsync(
-                () => _gitHoster.GetMyRepos(),
-                repos =>
+            ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
                 {
-                    _myReposLV.Items.Clear();
-                    foreach (var repo in repos)
+                    try
                     {
-                        var lvi = new ListViewItem { Tag = repo, Text = repo.Name };
-                        lvi.SubItems.Add(repo.IsAFork ? _strYes.Text : _strNo.Text);
-                        lvi.SubItems.Add(repo.Forks.ToString());
-                        lvi.SubItems.Add(repo.IsPrivate ? _strYes.Text : _strNo.Text);
-                        _myReposLV.Items.Add(lvi);
+                        await TaskScheduler.Default;
+
+                        var repos = _gitHoster.GetMyRepos();
+
+                        await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                        _myReposLV.Items.Clear();
+
+                        foreach (var repo in repos)
+                        {
+                            _myReposLV.Items.Add(new ListViewItem
+                            {
+                                Tag = repo,
+                                Text = repo.Name,
+                                SubItems =
+                                {
+                                    repo.IsAFork ? _strYes.Text : _strNo.Text,
+                                    repo.Forks.ToString(),
+                                    repo.IsPrivate ? _strYes.Text : _strNo.Text
+                                }
+                            });
+                        }
                     }
-                },
-                ex =>
-                {
-                    _myReposLV.Items.Clear();
-                    _helpTextLbl.Text = string.Format(_strFailedToGetRepos.Text, _gitHoster.Description) +
-                        "\r\n\r\nException: " + ex.Exception.Message + "\r\n\r\n" + _helpTextLbl.Text;
-                });
+                    catch (Exception e)
```

Comment:
:bulb: Use an exception filter to ignore cancellation-related exceptions (applies to other cases below as well)

Reply:
Other filters check a token for cancellation, but there's no token here. I cannot see how the operation would be cancelled. Could it? Should the filter exclude `TaskCanceledException` and/or `OperationCanceledException`?

Reply:
> I cannot see how the operation would be cancelled.

Yes, the extension method form of switching to main thread uses a `CancellationToken` to handle the disposed case.